### PR TITLE
[TF] minibatch-sgd

### PR DESCRIPTION
**Pull request by abhinavsp0730**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1264/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1264/

File: chapter_optimization/minibatch-sgd.md [hunk 1]
```diff
@@ -50,6 +50,22 @@ B = torch.randn(256, 256)
...
 
+```{.python .input}
+#@tab tensorflow
+%matplotlib inline
+from d2l import tensorflow as d2l
+import tensorflow as tf
+import numpy as np
+
+timer = d2l.Timer()
+# not using tf.xxx here because
```
Comment (terrytangyuan) [inline chapter_optimization/minibatch-sgd.md:61]: Use tf.Variable

File: d2l/mxnet.py [hunk 2]
```diff
@@ -2559,27 +2559,6 @@ def update_G(Z, net_D, net_G, loss, trainer_G):  #@save
...
-
-
-def load_data_bananas(batch_size, edge_size=256):
-    """Load the bananas dataset."""
-    data_dir = d2l.download_extract('bananas')
-    train_iter = image.ImageDetIter(
-        path_imgrec=os.path.join(data_dir, 'train.rec'),
-        path_imgidx=os.path.join(data_dir, 'train.idx'),
-        batch_size=batch_size,
-        data_shape=(3, edge_size, edge_size),  # The shape of the output image
```
Comment (terrytangyuan) [inline d2l/mxnet.py:2573]: Why these are removed?

**Comment by mli** [comment]
Job d2l-en/PR-1264/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1264/

File: chapter_optimization/minibatch-sgd.md (hunk 1 above)
Comment (abhinavsp0730) [reply chapter_optimization/minibatch-sgd.md:61]: Hey, @terrytangyuan I tried tf.Variables but got the same type of error 'ResourceVariable' object does not support item assignment

Comment (terrytangyuan) [reply chapter_optimization/minibatch-sgd.md:61]: Use assign() for tf variables

**Comment by mli** [comment]
Job d2l-en/PR-1264/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1264/

File: chapter_optimization/minibatch-sgd.md (hunk 1 above)
Comment (abhinavsp0730) [reply chapter_optimization/minibatch-sgd.md]: Done.

**Comment by abhinavsp0730** [comment]
Hey, @terrytangyuan please review.

**Comment by astonzhang** [comment]
Can you rebase onto master HEAD to pass the ci?

**Comment by mli** [comment]
Job d2l-en/PR-1264/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1264/

**Comment by astonzhang** [comment]
@abhinavsp0730 Can you keep each line of code within 78 chars?

**Comment by mli** [comment]
Job d2l-en/PR-1264/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1264/

**Comment by abhinavsp0730** [comment]
@astonzhang fixed the 78 char issue, please review now.

File: chapter_optimization/minibatch-sgd.md [hunk 3]
```diff
@@ -272,6 +348,40 @@ def train_ch11(trainer_fn, states, hyperparams, data_iter,
...
+      trainer_fn([w, b], [dw, db], states, hyperparams)
+      n += X.shape[0]
+      if n % 200 == 0:
+          timer.stop()
+          p = n/X.shape[0]/tf.data.experimental.cardinality(data_iter).numpy()
+          q = (d2l.evaluate_loss(net, data_iter, loss),)
+          animator.add(p, q)
+          timer.start()
+  print(f'loss: {animator.Y[0][-1]:.3f}, {timer.avg():.3f} sec/epoch')
+  return timer.cumsum(), animator.Y[0]
```
Comment (AnirudhDagar) [inline chapter_optimization/minibatch-sgd.md:382]: Please use appropriate tab spacing (4) for the function.

Comment (abhinavsp0730) [reply chapter_optimization/minibatch-sgd.md]: please check now.

**Comment by abhinavsp0730** [comment]
@astonzhang @AnirudhDagar I've fixed the tab issues. Also, there is 78 char issue in get_data_ch11( ) in pytorch. So, I fixed it too.

**Comment by mli** [comment]
Job d2l-en/PR-1264/8 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1264/

**Comment by mli** [comment]
Job d2l-en/PR-1264/10 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1264/
